lib/improver/nowcasting/accumulation.py

```diff
@@ -0,0 +1,209 @@
+            raise ValueError(msg)
+
+        cube_subset = int(cube_subset)
+
+        # We expect 1 more rates cubes than accumulations cubes, if we have
+        # more or less, a targetted accumulation_period will be incomplete.
+        unused_cubes = int(len(cubes) % cube_subset)
+        if unused_cubes != 1 and cube_subset != 1:
+            msg = ("The provided cubes result in a partial period given the "
+                   "specified accumulation_period, i.e. the number of cubes "
+                   "is insufficient to give a set of complete periods. Only "
+                   "complete periods will be returned.")
+            warnings.warn(msg)
+
+        cube_subsets = []
+        for index in range(0, len(cubes) - unused_cubes, cube_subset):
+            cube_subsets.append(cubes[index:index + cube_subset + 1])
+        return cube_subsets
+
+    def process(self, cubes):
+        """
+        Calculate period precipitation accumulations based upon precipitation
+        rate fields.
+
+        Args:
+            cubes (iris.cube.CubeList):
+                A cubelist containing input precipitation rate cubes.
+        Returns:
+            accumulation_cubes (iris.cube.CubeList):
+                A cubelist containing precipitation accumulation cubes where
+                the accumulation periods are determined by plugin argument
+                accumulation_period.
+        """
+        # Standardise inputs to expected units.
+        cubes = enforce_coordinate_units_and_dtypes(
+            cubes, ['time', 'forecast_reference_time', 'forecast_period'],
+            inplace=False)
+        enforce_diagnostic_units_and_dtypes(cubes)
+
+        # Sort cubes into time order and calculate intervals.
+        cubes, times = self.sort_cubes_by_time(cubes)
+        try:
+            time_interval, = np.unique(np.diff(times, axis=0))
+        except ValueError:
+            msg = ("Accumulation is designed to work with "
+                   "rates cubes at regualar time intervals. Cubes "
+                   "provided are unevenly spaced in time; time intervals are "
+                   "{}.".format(np.diff(times, axis=0)))
+            raise ValueError(msg)
+
+        cube_subsets = self.get_period_sets(time_interval, cubes)
+
+        accumulation_cubes = iris.cube.CubeList()
+        for cube_subset in cube_subsets:
+            # Handle accumulation periods equal to the rates cube time interval
+            # in which case the last subset does not get used.
+            if len(cube_subset) == 1:
+                continue
+
+            accumulation = 0.
+            for cube in cube_subset[0:-1]:
+                accumulation += cube.data * time_interval
+
+            # Use CubeCombiner to expand coordinate bounds.
+            cube_name = 'lwe_thickness_of_precipitation_amount'
+            accumulation_cube = CubeCombiner('add').process(
+                iris.cube.CubeList(cube_subset), cube_name,
```

Comment:
Why not just create a new cube here?  All the information is available on the input cubes and we've done the calculation in the accumulation plugin.  Metadata updates shouldn't be tied to unnecessary processing like this.

Reply:
I've now factored out the expand_bounds functionality which is what I was really using the cube combiner for.